Source/Bogus.Tests/FakerTests.cs

```diff
@@ -0,0 +1,237 @@
+
+      [Fact]
+      public void SkipConstructor_WithRule_Default_StrictMode_Not_Satisfied()
+      {
+         Assert.Throws<ValidationException>(() =>
+         {
+            _ = new Faker<NoParameterlessCtorClass>()
+               .SkipConstructor()
+               .RuleFor(x => x.Obj, x => null)
+               .Generate(1);
+         });
+      }
+
+      [Fact]
+      public void SkipConstructor_WithoutRules_Default_StrictMode_Not_Satisfied()
+      {
+         Assert.Throws<ValidationException>(() =>
+         {
+            _ = new Faker<NoParameterlessCtorClass>()
+               .SkipConstructor()
+               .Generate(1);
+         });
+      }
+
+      [Fact]
+      public void Not_Skipping_Constructor_Without_Parameterless_Ctor()
+      {
+         Assert.Throws<NoParameterlessCtorException<NoParameterlessCtorClass>>(() =>
+         {
+            _ = new Faker<NoParameterlessCtorClass>()
+             .Generate(1);
+         });
+      }
+
+      [Fact]
+      public void Skipping_Constructor_Overiding_StrictMode()
+      {
+
+         var faker = new Faker<NoParameterlessCtorClass>()
+          .SkipConstructor()
+          .StrictMode(false);
+
+         Assert.Contains(faker.StrictModes, (x => x.Value == false));
+         _ = faker.Generate(1);
+
+
+
+         Assert.Throws<ValidationException>(() =>
+         {
+            faker = new Faker<NoParameterlessCtorClass>()
+               .SkipConstructor()
+                .StrictMode(true);
+
+            Assert.Contains(faker.StrictModes, (x => x.Value == true));
+            _ = faker.Generate(1);
+
+         });
+
+      }
+
+
+      [Fact]
+      public void UseConstructor()
+      {
+         var faker = new Faker<NoParameterlessCtorClass>()
+            .UseConstructor(x => new NoParameterlessCtorClass(new Order(), 5));
+         faker.Generate(1);
+
+         var faker2 = new Faker<Order>()
+             .UseConstructor(x => new Order() { OrderId = 68 });
+         var order = faker2.Generate(1)[0];
+
+         Assert.Equal(68, order.OrderId);
+      }
+
+
+      [Fact]
+      public void UseConstructor_WithRule_Default_StrictMode_Not_Satisfied()
+      {
+         Assert.Throws<ValidationException>(() =>
+         {
+            _ = new Faker<NoParameterlessCtorClass>()
+               .UseConstructor(x => new NoParameterlessCtorClass(default, default))
+               .StrictMode(true)
+               .RuleFor(x => x.Obj, x => null)
+               .Generate(1);
+         });
+      }
+
+      [Fact]
+      public void UseConstructor_WithoutRules_Default_StrictMode_Not_Satisfied()
+      {
+         Assert.Throws<ValidationException>(() =>
+         {
+            _ = new Faker<NoParameterlessCtorClass>()
+                  .StrictMode(true)
+                  .UseConstructor(x => new NoParameterlessCtorClass(default, default))
+                  .Generate(1);
+         });
+      }
+
+
+      [Fact]
+      public void UseContructor_After_SkipConstructor()
```

Comment:
Is debatable if using both `UseConstructor` and `SkipConstructor` should be prohibited or not although it does not break anything.